hnn_core/tests/test_io.py

```diff
@@ -0,0 +1,130 @@
+import os.path as op
+from numpy.testing import assert_allclose
+from h5io import write_hdf5
+import pytest
+
+import hnn_core
+from hnn_core import read_network
+from hnn_core import simulate_dipole
+from hnn_core import read_params
+from hnn_core import jones_2009_model, law_2021_model, calcium_model
+
+hnn_core_root = op.dirname(hnn_core.__file__)
+params_fname = op.join(hnn_core_root, 'param', 'default.json')
+
+
+@pytest.mark.parametrize("network_model",
+                         [law_2021_model, calcium_model,
+                          jones_2009_model])
+def test_network_io(tmp_path, network_model):
+    # For simulation to be shorter(Discuss)
+    params = op.join(hnn_core_root, 'param', 'default.json')
+    if isinstance(params, str):
+        params = read_params(params)
+    params['N_pyr_x'] = 3
+    params['N_pyr_y'] = 3
+    params['celsius'] = 37.0
+    params['threshold'] = 0.0
+
+    # Instantiating network along with drives
+    net = network_model(params=params, add_drives_from_params=True)
+
+    # Adding bias
+    net.add_tonic_bias(cell_type='L2_pyramidal', amplitude=1.0)
+
+    # Test __eq__ method
+    net_copy = net.copy()
+    assert net_copy == net
+
+    # Test other not NotImplemented for Network Class
+    assert (net == "net") is False
```

Comment:
```suggestion
    with pytest.raises(NotImplemented):
        net == "net"
```
This is a more robust test for this error